src/main/java/com/liveramp/captain/daemon/CaptainJoblet.java

```diff
@@ -188,6 +189,11 @@ private void submitRequest(CaptainRequestConfig config) throws DaemonException {
       notifier.notify(subject, e, CaptainNotifier.NotificationLevel.ERROR);
       requestUpdater.cancel(id);
       return;
+    } catch (CaptainSubmissionException ce) {
+      String subject = String.format("%s: handle transient submission failed for request %s", CaptainAlertHelpers.getHostName(), id);
```

Comment:
This exception doesn't seem conceptually very different from the above `CaptainPersistorException`. I wonder if anyone is actually using that one. If not, perhaps we can repurpose it. If yes, we should choose a name for this new exception that better distinguishes it

Reply:
CaptainPersistorException is used when actual saving to database fails. CaptainSubmitRequestFailedException & CaptainStatusRetrieverFailedException sounds ok?